Tests/UnitTests/Errors/SignOutWarningViewModelTests.swift

```diff
@@ -43,7 +44,29 @@ extension SignOutWarningViewModelTests {
     
     func test_buttonAction() throws {
         XCTAssertFalse(didCallButtonAction)
+        XCTAssertEqual(mockAnalyticsService.eventsLogged.count, 0)
         sut.primaryButtonViewModel.action()
         XCTAssertTrue(didCallButtonAction)
+        XCTAssertEqual(mockAnalyticsService.eventsLogged.count, 1)
+        let event = LinkEvent(textKey: "app_extendedSignInButton",
+                              linkDomain: AppEnvironment.oneLoginBaseURL,
+                              external: .false)
+        XCTAssertEqual(mockAnalyticsService.eventsLogged, [event.name.name])
+        XCTAssertEqual(mockAnalyticsService.eventsParamsLogged["text"], event.parameters["text"])
```

Comment:
Please can we not check each of the parameters individually? This tightly couples our tests to the implementation of each event type, which we don't want to do. For example, if we were to then change the implementation of the event in the logging package, we would then break all our unit tests.

We should just check that `eventParamsLogged` equals `event.parameters` 🙂

Reply:
Sure can do! This copies how we've been testing `eventParamsLogged` in our other tests, do they need a tech debt ticket to correct given that it's technically outside the scope of this ticket?

Reply:
Yes, can we correct these ones here and then raise a ticket for the others please?

Reply:
Will do! Should `screenParamsLogged` also be updated to reflect the above changes, or is that fine being like `XCTAssertEqual(mockAnalyticsService.screenParamsLogged["title"], screen.parameters["title"])`?

Reply:
Yeah, I think we should do the same if we can. The tests should be flexible such that if we rename the parameters in the login package, then the tests here don't break - the tests in the logging package already cover that.

Reply:
I've updated both `eventParamsLogged` and `screenParamsLogged`. Is this change what you had envisioned?